src/paletteMachine.ts

```diff
@@ -23,14 +22,10 @@ export const paletteMachine = paletteModel.createMachine({
             };
             return Object.values(keybindings).some(Boolean);
           }
-          function eventRoseFromEditor(e: KeyboardEvent) {
-            const editorElement = document.querySelector(EDITOR_CLASSNAME);
-            return editorElement && editorElement.contains(e.target as Node);
-          }
           const eventHandler = (e: KeyboardEvent) => {
             if (
-              captureCommandPaletteKeybindings(e) &&
-              !eventRoseFromEditor(e)
+              !isInputLikeElement(e.target as HTMLElement) &&
```

Comment:
it means that the palette also won't be showed when we have focus on checkboxes, radios and similar - as the check inside this helper only checks the `tagName`. This seems fine though? Maintaining a list of allowlist/denylist of input types seemed a little bit painful for me so I've ignored this for the time being. Thoughts?

Reply:
I think it should show. What if we changed our approach to stopping propagation inside inputs, rather than checking if the event came from an input?

Reply:
I've considered stopping the propagation approach in the past but the problem with that is that then the "logic" for that is scattered all over the place and it's easy to regress on the correct behaviour. You need to keep an eye on this and be aware that this is even a thing - whereas this logic is centralized and when you add new components to your app you simply don't have to care about this issue.

I've done a summary for all `input` types to check out how they behave and what kind of a check could be implemented here:
| `input` type   | keyboard input                                                                        |
| -------------- | ------------------------------------------------------------------------------------- |
| button         | definitely doesn't accept text                                                        |
| checkbox       | definitely doesn't accept text                                                        |
| color          | it renders as a button - keydowns in the popup are ignored                            |
| date           | it only accepts numerical text - keydowns in the popup are ignored                    |
| datetime-local | it only accepts numerical text - keydowns in the popup are ignored                    |
| email          | definitely accepts text                                                               |
| file           | it renders as a button                                                                |
| hidden         | it doesn't render at all                                                              |
| image          | it acts as a graphical submit button                                                  |
| month          | it only accepts alphanumerical text - keydowns in the popup are ignored               |
| number         | it only accepts numerical text                                                        |
| password       | definitely accepts text                                                               |
| radio          | definitely doesn't accept text                                                        |
| range          | it only accepts arrows                                                                |
| reset          | it renders as a button                                                                |
| search         | definitely accepts text                                                               |
| submit         | it renders as a button                                                                |
| tel            | definitely accepts text                                                               |
| text           | definitely accepts text                                                               |
| time           | it only accepts numerical text and spacebar/enter - keydowns in the popup are ignored |
| url            | definitely accepts text                                                               |
| week           | it only accepts numerical text - keydowns in the popup are ignored                    |

The intention was not to list all meaningful keyboard interactions for those but to give a vague notion of what kind of an input is accepted by them. I was testing this here: https://codesandbox.io/s/kind-browser-9hrwz?file=/src/App.js

I think we only have two available strategies here to allow those interactions on some inputs - it's either to allow them for specific input types or to disallow them for the complementary values. The first one seems to be conceptually slightly better - if we miss something or a new input type gets added to the platform the worst thing that happens is that the shortcut won't work in those cases. Using the opposite strategy bears the risk of interfering with the user input - which is a bad UX.

It's really tempting to go with the second approach though as the list of input types that would have to be used for it is much shorter.

It's also worth noting that this simple approach only works for certain global shortcuts - we could imagine a situation in other apps when a particular shortcut would clash with, for example, numerical inputs. In such a situation I imagine one would have to implement different strategies for different in-app shortcuts that would account for different input types. This doesn't apply to us though - at least not right now.

I've implemented adjustments here: https://github.com/statelyai/xstate-viz/pull/208/commits/2525df88c888c68d61eaec6061610a29b4236ccc